### ISSUE TITLE
Pre-exploration in the unseen environment

### ISSUE THREAD

**Issue by chijames** [airsplay/R2R-EnvDrop#4]
Hi,

Can you please share the augmented path file in the unseen environment? If I understand correctly, the aug_paths.json is only for the back translation in the seen environment. If it is not possible, some statistics about the unseen augmented path would also be very helpful. Thanks.

**Comment by airsplay** [comment]
Sure. [test](aug_paths_test.json) and [valid unseen](aug_paths_unseenvalid.json). Test data and valid unseen data have been carefully removed from these aug paths.  For these experiments, I actually only allow the agent to explore the test environment but not give it testing instructions (as in RCM).

I have not tested the performance with new PyTorch version but the result should be reproducible by replacing ```--aug tasks/R2R/data/aug_paths.json``` in ```bt_envdrop.bash``` with appropriate json files (especially on unseen valid because you could see the local evaluation). I would update github after verification. Could you please leave this issue open until then?

**Comment by chijames** [comment]
Thanks for the response. However, the links you provided seem to be broken? I will leave this issue open until then.

**Comment by chijames** [comment]
Thanks for the timely response, I will leave this issue open. By "not give it testing instructions", you mean the instructions will be generated by the trained speaker model with environmental dropout, right? Thanks.

**Comment by airsplay** [comment]
Yep. And it would never touch the paths/instructions in testing data and validation unseen data.

**Comment by chijames** [comment]
May I know if there are some rules you use for generating augmented paths? Since I want to sample more paths to see the upper bound of the pre-exploration method. Currently, I randomly sample two viewpoints with their distance larger than 5 and a start heading angle. Thanks.

**Comment by airsplay** [comment]
Sorry for late replying (due to ACL). 

The file is generated by exploiting all viewpoint pairs which have action length from 4~6. I also exclude the val/test data. Thus it is a complete set of all available short-distance paths. The initial heading angle are randomly sampled. I visualize the headings in training data and I believe that the initial headings in training are uniformly sampled.

If you want to verify the upper bound, I suggest to try:
1. Sampling more initial headings.
2. Generate more instructions for each paths. (currently one for each)

I am also notified (by Peter) that the speaker model trained with PyTorch 1.0 might be weaker than with PyTorch 0.4. Since the pre-exploring results highly depends on the performance of speaker, I doubt whether the results are still the same. Have you achieved any similar result by adding pre-exploring paths? If not, I would definitely take more time on fixing the speaker issue.

**Comment by yestinl** [comment]
> Sure. [test](http://www.cs.unc.edu/~airsplay/aug_paths_test.json) and [valid unseen](http://www.cs.unc.edu/~airsplay/aug_paths_unseenvalid.json). Test data and valid unseen data have been carefully removed from these aug paths. For these experiments, I actually only allow the agent to explore the test environment but not give it testing instructions (as in RCM).
> 
> I have not tested the performance with new PyTorch version but the result should be reproducible by replacing `--aug tasks/R2R/data/aug_paths.json` in `bt_envdrop.bash` with appropriate json files (especially on unseen valid because you could see the local evaluation). I would update github after verification. Could you please leave this issue open until then?

May I know how to allow the agent to explore the test environment but not give it testing instructions?

**Comment by airsplay** [comment]
It is completed via the trick of back-translation. The paths are first randomly sampled and the instructions are then generated from the speaker.

**Comment by yestinl** [comment]
Thanks for timely reply! I just want to know how to implement.

**Comment by yestinl** [comment]
> It is completed via the trick of back-translation. The paths are first randomly sampled and the instructions are then generated from the speaker.

So should I use the speaker module to generate the instruction of test split? And then train them as augmentation data?

**Comment by amoudgl** [comment]
@airsplay Does `aug_paths.json` also contain original R2R train + val seen data?
